src/Control/Algebra/Choose/Church.hs

```diff
@@ -63,7 +62,8 @@ instance MonadTrans ChooseC where
   lift m = ChooseC (\ _ leaf -> m >>= leaf)
   {-# INLINE lift #-}
 
-instance (Algebra sig m, Effect sig) => Algebra (Choose :+: sig) (ChooseC m) where
+instance (Algebra m, Effect (Signature m)) => Algebra (ChooseC m) where
+  type Signature (ChooseC m) = Choose :+: Signature m
```

Comment:
Instance heads become a bit tidier, and `MultiParamTypeClasses` gets replaced with `TypeFamilies`.